plugins/inputs/modbus/README.md

```diff
@@ -85,15 +85,16 @@ Registers via Modbus TCP or Modbus RTU/ASCII.
   ## data_type   - INT16, UINT16, INT32, UINT32, INT64, UINT64,
   ##               FLOAT32-IEEE, FLOAT64-IEEE (the IEEE 754 binary representation)
   ##               FLOAT32, FIXED, UFIXED (fixed-point representation on input)
-  ## scale       - the final numeric variable representation
+  ## scale       - the final numeric variable representation will be (scale × variable + shift)
+  ## shift       - the final numeric variable representation will be (scale × variable + shift)
```

Comment:
I initially also called it offset but, in Modbus, the offset usually refers to a change to the address in the Modbus message and I think it can be confusing (there are already variables called offset in the code).
Also, if we are to use namings used in electronics, then `scale` should be changed to `gain` but this would make former configuration files incompatible.

Reply:
Thanks for your feedback @srebhan 
Many of the things you pointed out in the review come from the fact that I had the *register* configuration in mind when I wrote the code, whereas some functions were written with the *request* configuration. I believe it is a problem that both these configurations have different behaviours.
For instance, 
* request has *omit* and *output* but register not
* register accepts "AB" byte order (as it should be for INT16) but request does not
* register expects *address* to be an array but request wants a single int
* other checks are also different

I believe this makes the code more difficult to maintain and makes the user's life more difficult. Do you want me to try to modify the code more so that both syntaxes converge?

Reply:
No please don't. This is by intention as we need to keep backward-compatibility with old config-files. Therefore, the configuration schemes differ quite a bit. I don't think this complicates code as the _config parsing code_ is separated and the core operates on a common scheme. Furthermore, please make changes to the config _style_ a separate discussion (maybe in an issue)...